modules/sdk-coin-tia/src/lib/transaction.ts

```diff
@@ -0,0 +1,33 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { fromHex, toBase64 } from '@cosmjs/encoding';
+
+import tiaUtils from './utils';
+import { CosmosTransaction } from '@bitgo/abstract-cosmos';
+
+export class TiaTransaction extends CosmosTransaction {
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  /**
+   * Sets this transaction payload
+   * @param rawTransaction raw transaction in base64 encoded string
+   */
+  enrichTransactionDetailsFromRawTransaction(rawTransaction: string): void {
+    if (tiaUtils.isValidHexString(rawTransaction)) {
+      this.cosmosLikeTransaction = tiaUtils.deserializeTiaTransaction(toBase64(fromHex(rawTransaction)));
+    } else {
+      this.cosmosLikeTransaction = tiaUtils.deserializeTiaTransaction(rawTransaction);
+    }
+    if (this.cosmosLikeTransaction.signature) {
+      this.addSignature(Buffer.from(this.cosmosLikeTransaction.signature).toString('hex'));
+    }
+    const typeUrl = this.cosmosLikeTransaction.sendMessages[0].typeUrl;
+    const transactionType = tiaUtils.getTransactionTypeFromTypeUrl(typeUrl);
+
+    if (transactionType === undefined) {
+      throw new Error('Transaction type is not supported ' + typeUrl);
+    }
+    this.transactionType = transactionType;
+  }
```

Comment:
Seems pretty similar to [ATOM](url). If this is the case for osmosis/other chains, I'd say we'll need this in the common class instead (since we're anyway utilising `cosmosLikeTransaction`, which is a super type)

Reply:
apart from a few utility functions rest of the code looks exactly same for all these chains. I think we need to find some better way of doing it